cmd/promtool/main_test.go

```diff
@@ -203,3 +204,40 @@ func TestCheckTargetConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestAuthorizationConfig(t *testing.T) {
+	cases := []struct {
+		name   string
+		file   string
+		err    string
+		errWin string
+	}{
+		{
+			name:   "authorization_credentials_file.bad",
+			file:   "authorization_credentials_file.bad.yml",
+			err:    `error checking authorization credentials file "/random/file/which/does/not/exist.yml": stat /random/file/which/does/not/exist.yml: no such file or directory`,
+			errWin: `error checking authorization credentials file "testdata\\random\\file\\which\\does\\not\\exist.yml": CreateFile testdata\random\file\which\does\not\exist.yml: The system cannot find the path specified.`,
+		},
+		{
+			name: "authorization_credentials_file.good",
+			file: "authorization_credentials_file.good.yml",
+			err:  "",
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := checkConfig(false, "testdata/"+test.file)
+
+			if runtime.GOOS != "windows" && test.err != "" {
+				require.Equal(t, test.err, err.Error(), "Expected error %q, got %q", test.err, err.Error())
+				return
+			}
+			if runtime.GOOS == "windows" && test.errWin != "" {
+				require.Equal(t, test.errWin, err.Error(), "Expected error %q, got %q", test.err, err.Error())
+				return
+			}
```

Comment:
While accounting for operating systems' different error messages is nice, I think it would be simpler to just check if the error contains the part of the message we generate (`error checking authorization credentials file`).

```suggestion
			if test.err != "" {
				require.Contains(t, err.Error(), t.err, "Expected error %q to contain %q", err.Error(), test.err)
				return
			}
```